core/embed/rust/src/ui/model_tt/component/address_details.rs

```diff
@@ -7,32 +7,38 @@ use crate::{
             text::paragraphs::{
                 Paragraph, ParagraphSource, ParagraphStrType, ParagraphVecShort, Paragraphs, VecExt,
             },
-            Component, Event, EventCtx, Never, Paginate, Qr,
+            Component, Event, EventCtx, Paginate, Qr,
         },
         geometry::Rect,
     },
 };
 
-use super::{theme, Frame};
+use super::{theme, Frame, FrameMsg};
+
+const MAX_XPUBS: usize = 16;
 
 pub struct AddressDetails<T> {
     qr_code: Frame<Qr, T>,
     details: Frame<Paragraphs<ParagraphVecShort<T>>, T>,
     xpub_view: Frame<Paragraphs<Paragraph<T>>, T>,
-    xpubs: Vec<(T, T), 16>,
+    xpubs: Vec<(T, T), MAX_XPUBS>,
+    xpub_pages: Vec<u8, MAX_XPUBS>,
```

Comment:
Renamed the field 22b73b8f2ad7a47e1cd168c1426cc7bf9328f7de. Having the field be usize makes the struct 48 bytes smaller, hard to say whether it's a good justification. I've at least got rid of the `as usize` casts and replaced them with `.into()` since it is defined for `u8->usize`.